src/Common/src/System/IO/RowConfigReader.cs

```diff
@@ -82,7 +82,11 @@ public bool TryGetNextValue(string key, out string value)
             // after. This is the format of most "row-based" config files in /proc/net, etc.
             int afterKey = keyIndex + key.Length;
             int endOfLine = _buffer.IndexOf(Environment.NewLine, afterKey, _comparisonKind);
-            Debug.Assert(endOfLine != -1, "RowConfigReader needs a newline after the key, and one was not found.");
+            if (endOfLine == -1)
+            {
+                // There may not be a newline after this key, if we've reached the end of the file.
+                endOfLine = _buffer.Length - 1;
+            }
```

Comment:
Yeah, it will behave badly in that scenario. Right now I'm assuming a base level of "well-formatted-ness", since the files we are parsing are either OS-generated or assumed to be in a particular format by the OS if edited by hand. But I would prefer if this type could handle such cases, since it should be usable in general cases.

I think we can avoid your particular issue by using a different overload of `LastIndexOf`. We can error out if it turns out there's not actually a value present for that particular key.